### DISCUSSION TITLE
for VTK data, it incluldes three components. how can choose one component for plot

### DISCUSSION THREAD

**Discussion by subenyu** [pyvista/pyvista#1662] [Q&A]
for VTK data, it incluldes three solid color ,mark, Resisitivity,  Resisitivity_log(10). I just to use  Resisitivity_log(10)  to draw the figure.
How can do it?

ms = pv.read(rootpath+'/resistivity.vtk')
cmap = plt.cm.get_cmap("jet")
ms.plot(cmap=cmap,show_bounds=[2000, 2000],screenshot=rootpath+'/data/vol.png',\
window_size=[2000, 2000],opacity=1.0)

**Comment by MatthewFlamm** [comment]
I'm confused by the question.  Do you have multiple scalar arrays and want to plot one, or a multi component array and want to plot one component?

**Comment by subenyu** [comment]
@MatthewFlamm 
Thank you very much for your reply and sorry for my poor description. My reak questtion is "a multi component array and want to plot one component?" by pyvista.
In fact, myfile.vtk inclues two component array, Marks, logres but i only want the figure of  logres. 
Thank you very much.
 ```py
import os
import matplotlib.pyplot as plt
import pyvista as pv
rootpath = os.getcwd()
p = pv.Plotter(rootpath+'/myfile.vtk')
ms = pv.read(rootpath+'/myfile.vtk')
cmap = plt.cm.get_cmap("jet")
ms.plot(cmap=cmap,show_bounds=[2000, 2000],screenshot=rootpath+'/data/vol.png',\
window_size=[2000, 2000],opacity=1.0)
```
[myfile.zip](https://github.com/pyvista/pyvista/files/7158196/myfile.zip)

**Reply by MatthewFlamm** [reply]
Have you tried:

```
ms.plot(scalars='logres', cmap=cmap,show_bounds=[2000, 2000],screenshot=rootpath+'/data/vol.png',
        window_size=[2000, 2000],opacity=1.0)
```

**Reply by subenyu** [reply]
Thank you very much and it work very good. And sorry have another question. if i want to generate colorbar and colorbar can be shifted? How can i do it ?
Thank you very much.

**Reply by MatthewFlamm** [reply]
See https://docs.pyvista.org/examples/02-plot/scalar-bars.html?highlight=scalar_bar
